### PR TITLE
finagle-redis: Add RPOPLPush Command

### PR DESCRIPTION
**Pull request by martinffx**
# Problem

In order to implement a reliable queueing mechanism using redis
you need this command.

# Solution

Added command to `ListCommands`

### CODE REVIEW

File: finagle-redis/src/main/scala/com/twitter/finagle/redis/protocol/commands/Lists.scala [hunk 1]
```diff
@@ -62,6 +62,10 @@ case class LTrim(key: Buf, start: Long, end: Long) extends ListRangeCommand {
   def name: Buf = Command.LTRIM
 }
 
+case class RPopLPush(source: Buf, dest: Buf) extends MoveCommand {
```
Comment (kevinoliver) [inline finagle-redis/src/main/scala/com/twitter/finagle/redis/protocol/commands/Lists.scala]: i /think/ `dest` should be named `destination` in order to satisfy the `MoveCommand` trait.

**Review by kevinoliver — CHANGES_REQUESTED** [review]
looks good. i think it needs one minor fix to compile.

did you test this locally?

**Comment by martinffx** [comment]
I'm not sure how to test it locally?

```finagle/finagle-redis add-rpoplpush-command
❯ pants test

13:16:47 00:00 [main]
               (To run a reporting server: ./pants server)
13:16:48 00:01   [setup]
13:16:48 00:01     [parse]
               Executing tasks in goals: bootstrap -> imports -> unpack-jars -> unpack-wheels -> deferred-sources -> native-compile -> link -> jvm-platform-validate -> gen -> resolve -> compile -> pyprep -> resources -> test
13:16:48 00:01   [bootstrap]
13:16:48 00:01     [substitute-aliased-targets]
13:16:48 00:01     [jar-dependency-management]
13:16:48 00:01     [bootstrap-jvm-tools]
13:16:48 00:01     [provide-tools-jar]
13:16:48 00:01   [imports]
13:16:48 00:01     [ivy-imports]
13:16:48 00:01   [unpack-jars]
13:16:48 00:01     [unpack-jars]
13:16:49 00:02   [unpack-wheels]
13:16:49 00:02     [unpack-wheels]
13:16:49 00:02   [deferred-sources]
13:16:49 00:02     [deferred-sources]
13:16:49 00:02   [native-compile]
13:16:49 00:02     [conan-prep]
13:16:49 00:02     [conan-fetch]
13:16:49 00:02     [c-for-ctypes]
13:16:49 00:02     [cpp-for-ctypes]
13:16:49 00:02   [link]
13:16:49 00:02     [shared-libraries]
13:16:49 00:02   [jvm-platform-validate]
13:16:49 00:02     [jvm-platform-validate]
13:16:49 00:02   [gen]
13:16:49 00:02     [antlr-java]
13:16:49 00:02     [antlr-py]
13:16:49 00:02     [jaxb]
13:16:49 00:02     [protoc]
13:16:49 00:02     [ragel]
13:16:49 00:02     [thrift-java]
13:16:49 00:02     [thrift-py]
13:16:49 00:02     [grpcio-prep]
13:16:49 00:02     [grpcio-run]
13:16:49 00:02     [wire]
13:16:49 00:02   [resolve]
13:16:49 00:02     [ivy]
13:16:49 00:02     [coursier]
13:16:49 00:02   [compile]
13:16:49 00:02     [compile-jvm-prep-command]
13:16:49 00:02       [jvm_prep_command]
13:16:49 00:02     [compile-prep-command]
13:16:49 00:02     [compile]
13:16:49 00:02     [rsc]
13:16:49 00:02     [zinc]
13:16:49 00:02     [javac]
13:16:49 00:02   [pyprep]
13:16:49 00:02     [interpreter]
13:16:49 00:02     [build-local-dists]
13:16:49 00:02     [requirements]
13:16:49 00:02     [sources]
13:16:49 00:02   [resources]
13:16:49 00:02     [prepare]
13:16:49 00:02     [services]
13:16:49 00:02   [test]
13:16:49 00:02     [test-jvm-prep-command]
13:16:49 00:02       [jvm_prep_command]
13:16:49 00:02     [test-prep-command]
13:16:49 00:02     [test]
13:16:50 00:03     [pytest-prep]
13:16:50 00:03     [pytest]
13:16:50 00:03     [junit]
               Waiting for background workers to finish.
13:16:50 00:03   [complete]
               SUCCESS```

When I run `pants` it doesn't seem to do anything...

File: finagle-redis/src/main/scala/com/twitter/finagle/redis/protocol/commands/Lists.scala (hunk 1 above)
Comment (martinffx) [reply finagle-redis/src/main/scala/com/twitter/finagle/redis/protocol/commands/Lists.scala]: done

**Comment by kevinoliver** [comment]
> I'm not sure how to test it locally?
> When I run `pants` it doesn't seem to do anything...

unfortunately we have 2 build systems. pants is used internally at twitter, sbt is used for our oss builds.

take a look at [contributing](https://github.com/twitter/finagle/blob/develop/CONTRIBUTING.md) for how to build/test.

**Comment by martinffx** [comment]
So it's not exactly clear how to run the tests to me... I assume `sbt test` in the root dir.

https://github.com/twitter/finagle/blob/develop/CONTRIBUTING.md

There's lots of stuff on how to test but not how to run them or that you use sbt.

**Comment by martinffx** [comment]
I haven't added any tests for this change but I see there aren't really any tests for the `ListCodecSuite`, what sort of tests should I be adding?

**Comment by kevinoliver** [comment]
> So it's not exactly clear how to run the tests to me... I assume `sbt test` in the root dir.

good point! we don't mention that in the docs. and yes, `./sbt test` is right.

> I haven't added any tests for this change but I see there aren't really any tests for the ListCodecSuite, what sort of tests should I be adding?

i don't know the redis suite well. at a minimum, manual verification.

**Comment by yufangong** [comment]
Hey @martinffx, if you have manually verified this command, let us know! Thanks!

**Comment by martinffx** [comment]
I messed up the email on a couple of commits, I rebased but need to push to a new branch: https://github.com/twitter/finagle/pull/767

I'm closing this PR in favour of the new PR.

**Pull request closed without merging**